### DISCUSSION TITLE
Another GUI for sd.cpp - sd.cpp.gui.wx

### DISCUSSION THREAD

**Discussion by fszontagh** [leejet/stable-diffusion.cpp#255] [Show and tell]
Hi!

git:
https://github.com/fszontagh/sd.cpp.gui.wx

precompiled binaries packed into an installer (windows, debian .deb packages): https://github.com/fszontagh/sd.cpp.gui.wx/releases

The gui when starting up, auto detects which backend is usable. (CUDA, ROCm, or CPU avx variations). Of course, it's possible to force the backend with startup parameter:

> "-cuda" to force to load cuda
> "-rocm" to force to load rocm
> "-avx"
> "-avx2"
> "-avx512"
> "-allow-multiple" to allow multiple instances to run. Eg. if you wamt to start a cuda and an avx512 instance too:
> sd.gui.exe -cuda -allow-multiple
> sd.gui.exe -avx512 -allow-multiple
> The first instance no need to start with the "-allow-multiple" parameter to run another new one.

The UI is simple as is, written in C++ using wxWidgets with small footprint.

Some features: 

- job queue: you can start more image generation, but only one running at once. Pause / start, reload jobs (right click on a jobitem in the list)
![image](https://github.com/leejet/stable-diffusion.cpp/assets/51741446/d9e1ba84-d72a-442d-a941-1c8e919535ea)

- presets: you can save / reload some generation parameters
![image](https://github.com/leejet/stable-diffusion.cpp/assets/51741446/325773b1-ec51-46f4-a61e-782ec5580ef5)
![image](https://github.com/leejet/stable-diffusion.cpp/assets/51741446/9537f8de-cbea-4570-b110-e4db0559960e)

- simple model management: list local models, load model info from civitai (right click on model)
![image](https://github.com/leejet/stable-diffusion.cpp/assets/51741446/77fa3cbb-d6bc-48e1-a720-8b1624e30db3)
_On the preview images, it's possible to right click on it, and load the compatble parameters from the image._ 

- civitai client, a simple civitai model browser and downloader
![image](https://github.com/leejet/stable-diffusion.cpp/assets/51741446/19b57fff-5f1d-45b4-8863-70f47678d124)
![image](https://github.com/leejet/stable-diffusion.cpp/assets/51741446/7d40657e-f8b8-4eb8-8215-f5e22ee152ee)

- Generateion parameters are stored into the generated images, compatible with sd.next
- You can reload the generation parameters from an already generated image with drag & drop
... and more ... 

*screenshots made under windows 11. The GTK gui visual is differs

**Comment by clort81** [comment]
I like this!  Great feature set.  Except it seems crazy to use the app to download models.
You added to stable-diffusion.cpp multiple backends in one executeable? Is that a PR here?
